File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let rev = 0;

    while (x !== 0) {
        const rem = x % 10;
        rev = rev * 10 + rem;
        x = (x / 10) | 0;
    }

    if (rev < -(2 ** 31) || rev > 2 ** 31 - 1) {
        return 0;
    }

    return rev;
};

File: 7-reverse-integer/reverse-integer.go
func reverse(x int) int {
    rev := 0

    for x != 0 {
        rem := x % 10
        rev = rev*10 + rem
        x /= 10
    }

    if rev < -(1<<31) || rev > (1<<31)-1 {
        return 0
    }

    return rev
}
